### ISSUE TITLE
ImportError: No module named google.protobuf

### ISSUE THREAD

**Issue by pswpswpsw** [tensorflow/tensorflow#6341]
**Problem description**
This issue happens randomly. For same machine, same CPU,works randomly. I run it for 10 times, 6 times succeed, 4 times fail. Same code is running for every time. The code is 
```
import tensorflow as tf
# with tf.device('/cpu:0'):
a = tf.constant([1.0, 2.0, 3.0, 4.0, 5.0, 6.0], shape=[2, 3], name='a')
b = tf.constant([1.0, 2.0, 3.0, 4.0, 5.0, 6.0], shape=[3, 2], name='b')
c = tf.matmul(a, b)
with tf.Session(config=tf.ConfigProto(log_device_placement=True)) as sess:
	print sess.run(c)
```

**System information**
Information of cluster:
Basically our cluster is made of IBM Power System S822LC for commercial computing

    OS: Red Hat Enterprise Linux Server release 7.2 (Maipo)
    TF version: 0.10.0rc0
    protobuf: 3.0.0b2
    Python: 2.7.5

**Error**
```
>>> import os
>>> cwd = os.getcwd()
>>> cwd
'/gpfs/gpfs0/groups/duraisamy/shawnpan/test/script'
>>> import tensorflow
import tensorflow
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.so.5.1.5 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcurand.so.8.0 locally
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/gpfs/gpfs0/software/rhel72/packages/tensorflow_gpu/tensorflow/__init__.py", line 23, in <module>
    from tensorflow.python import *
  File "/gpfs/gpfs0/software/rhel72/packages/tensorflow_gpu/tensorflow/python/__init__.py", line 58, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "/gpfs/gpfs0/software/rhel72/packages/tensorflow_gpu/tensorflow/python/__init__.py", line 52, in <module>
    from tensorflow.core.framework.graph_pb2 import *
  File "/gpfs/gpfs0/software/rhel72/packages/tensorflow_gpu/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
    from google.protobuf import descriptor as _descriptor
ImportError: No module named google.protobuf


Error importing tensorflow.  Unless you are using bazel,
you should not try to import tensorflow from its source directory;
please exit the tensorflow source tree, and relaunch your python interpreter
from there.
```

**Comment by yaroslavvb** [comment]
Could it be an issue with the filesystem? (ie, Python is not able to import google.protobuf). You could perhaps debug by running distributed test script which does "import google.protobuf" and nothing else and seeing how often that fails

**Comment by martinwicke** [comment]
I believe this is an instance of https://github.com/google/protobuf/issues/1296. Which is annoying, but there's not much we can do about in in TensorFlow. 

There's a workaround in that thread, but it's some work for sysadmins that really should be inside protobuf.

**Comment by pswpswpsw** [comment]
problem resolved. it turns out to be googleprobuf corrupted on some nodes.

**Comment by bmatichuk** [comment]
On Mac OS - Installing tensorflow 1.3 - it will automatically remove other protobuf installs and install protobuf 3.4. However, this does not work and neither does installing or downgrading to any other protobuf version.

However I found a solution.
Not sure why this works - but on Mac OS this solved it.

    pip install google

**Comment by k-lam** [comment]
I install tensorflow with virtualenv.
pip uninstall protobuf
pip uninstall google
pip install google
pip install protobuf
That is ok,but I don't know why

**Comment by joe32140** [comment]
pip install google doesn't work for me
I just
pip install google-cloud
and it works

**Comment by keerthanpg** [comment]
pip install google-cloud worked for me as well! :)

**Comment by fangbq** [comment]
dyld: warning, LC_RPATH $ORIGIN/../../_solib_darwin_x86_64/_U_S_Stensorflow_Spython_C_Upywrap_Utensorflow_Uinternal.so___Utensorflow in /usr/local/lib/python2.7/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so being ignored in restricted program because it is a relative path
after installed google and google-cloud, I still got this:

```
Traceback (most recent call last):
  File "helloworld.py", line 1, in <module>
    import tensorflow as tf
  File "/usr/local/lib/python2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "/usr/local/lib/python2.7/site-packages/tensorflow/python/__init__.py", line 52, in <module>
    from tensorflow.core.framework.graph_pb2 import *
  File "/usr/local/lib/python2.7/site-packages/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
    from google.protobuf import descriptor as _descriptor
ImportError: No module named google.protobuf
```

**Comment by fangbq** [comment]
after installed google and google-cloud, I still got this:

Traceback (most recent call last):
  File "helloworld.py", line 1, in <module>
    import tensorflow as tf
  File "/usr/local/lib/python2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "/usr/local/lib/python2.7/site-packages/tensorflow/python/__init__.py", line 52, in <module>
    from tensorflow.core.framework.graph_pb2 import *
  File "/usr/local/lib/python2.7/site-packages/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
    from google.protobuf import descriptor as _descriptor
ImportError: No module named google.protobuf

**Comment by fangbq** [comment]
after installed google and google-cloud, I still got this:

```
Traceback (most recent call last):
  File "helloworld.py", line 1, in <module>
    import tensorflow as tf
  File "/usr/local/lib/python2.7/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "/usr/local/lib/python2.7/site-packages/tensorflow/python/__init__.py", line 52, in <module>
    from tensorflow.core.framework.graph_pb2 import *
  File "/usr/local/lib/python2.7/site-packages/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
    from google.protobuf import descriptor as _descriptor
ImportError: No module named google.protobuf
```

**Comment by malithakabir** [comment]
the __init__.py file is missing in site-packages/google directory. creating an empty __init__.py file there should work. thanks. @fangbq

**Comment by cryptox31** [comment]
On the Power System I faced the same error while installing tensorflow 1.5
Surprisingly, even though the protobuf module was there in the system, python did not pick it up.

After installing google and google-cloud with the following:
```
pip install google-cloud
pip install google
```
it worked fine.

**Comment by owenbrown** [comment]
pip install google-cloud worked for our windows dev who was having problems.

**Comment by songmeixu** [comment]
the __init__.py file is missing, save my time thanks! @malithakabir

**Comment by loretoparisi** [comment]
There is no need to install `google-cloud` or `google`. Neither the missing`__init__.py` will solve this issue!

**Comment by GeorMurk** [comment]
on my end, none of these solutions were working...

**Comment by YanliuWang** [comment]
pip install google-cloud-bigquery works for me , I am using windows, python 3.5 and tensorflow-gpu 1.4

**Comment by alexforever86** [comment]
Finally installed with conda
`conda install protobuf`

**Comment by manjotpahwa** [comment]
Do not install the google package since that will not solve your problem. You might have to uninstall it since it conflicts with the protobuf namespace. It's from a some other developer and not Google official!

Name: google
Version: 2.0.2
Summary: Python bindings to the Google search engine.
Home-page: http://breakingcode.wordpress.com/
Author: Mario Vilas
Author-email: mvilas@gmail.com
License: UNKNOWN
Requires: beautifulsoup4

**Comment by zwazel** [comment]
I'm following [this guide](https://jkjung-avt.github.io/build-tensorflow-1.12.2/) and every step worked fine except of the last one, When i try to run the benchmark i get this error. 

```
Traceback (most recent call last):
  File "scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py", line 25, in <module>
    import tensorflow as tf
  File "/home/zwazel/.local/lib/python3.6/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "/home/zwazel/.local/lib/python3.6/site-packages/tensorflow/python/__init__.py", line 52, in <module>
    from tensorflow.core.framework.graph_pb2 import *
  File "/home/zwazel/.local/lib/python3.6/site-packages/tensorflow/core/framework/graph_pb2.py", line 6, in <module>
    from google.protobuf import descriptor as _descriptor
ModuleNotFoundError: No module named 'google.protobuf'
```

I tried all of the above things and nothing worked. Can someone help me or should i open a new issue for this?

**Comment by craftsliu** [comment]
> I install tensorflow with virtualenv.
> pip uninstall protobuf
> pip uninstall google
> pip install google
> pip install protobuf
> That is ok,but I don't know why

good job,I also solve it in this way .
enviroment: Ubuntu 1.18

**Comment by Flamefire** [comment]
Again: Double-check that there is no `__init__.py` in your site-packages/google folder or delete it if there is. Easier than the uninstall dance which may or may not work. BTW: You probably do NOT want to install the "google" package: That is the Python bindings to google search engine. Why would you want that in this context? Oh and it trashes your namespace s mentioned in https://github.com/tensorflow/tensorflow/issues/6341#issuecomment-553769366

**Comment by gangmuk** [comment]
> Finally installed with conda
> `conda install protobuf`

It was the solution in my case. Thank you. pip was the problem. I am using conda environment. Use conda install!

**Comment by henrythung** [comment]
> I install tensorflow with virtualenv.
> pip uninstall protobuf
> pip uninstall google
> pip install google
> pip install protobuf
> That is ok,but I don't know why

This works for me too.
Before uninstall, the location of protobuf is \site-packages\protobuf-4.0.0rc2-py3.8.egg\google\protobuf, after reinstall, the location is changed to site-packages\google\protobuf. Probably this solve the problem.

**Comment by Rishit-dagli** [comment]
> I install tensorflow with virtualenv.
> pip uninstall protobuf
> pip uninstall google
> pip install google
> pip install protobuf
> That is ok,but I don't know why

This does work for me, but is indeed quite odd

**Comment by priyanksharma7** [comment]
> Again: Double-check that there is no `__init__.py` in your site-packages/google folder or delete it if there is. Easier than the uninstall dance which may or may not work. BTW: You probably do NOT want to install the "google" package: That is the Python bindings to google search engine. Why would you want that in this context? Oh and it trashes your namespace s mentioned in [#6341 (comment)](https://github.com/tensorflow/tensorflow/issues/6341#issuecomment-553769366)

I can confirm that removing `__init__.py ` does work! I am working on Jetson Nano with the latest JetPack 4.5.1 and installed Tensorflow 2.5.0 and protobuf 3.17.3. In my case, I use a virtual environment called `cv` so I ran the following commands:
cd ~/cv/lib/python3.6/site-packages/protobuf-3.17.3-py3.6-linux-aarch64.egg/google/
rm `__init__.py`

**Comment by Pogayo** [comment]
> > Finally installed with conda
> > `conda install protobuf`
> 
> It was the solution in my case. Thank you. pip was the problem. I am using conda environment. Use conda install!

Installing using conda worked for me too when I had a conda environment.

**Comment by babinos87** [comment]
For me it works when I create an `__init__.py` file inside `<venv_location>/lib/python3.10/site-packages/protobuf-4.21.11-py3.10.egg/google`.

But I don't think this is an acceptable solution. I shouldn't have to create any files manually. Other things mentioned (e.g. re-installing packages) do not work for me.

All I do is to install `google-ads`, which installs all required packages. Not a tensor flow issue, but it seems a google one.
